chia/wallet/did_wallet/did_wallet.py

```diff
@@ -141,18 +149,19 @@ async def create_new_did_wallet_from_recovery(
         wallet_state_manager: Any,
         wallet: Wallet,
         filename: str,
-        name: str = None,
+        name: str = "DID Wallet",
     ):
         self = DIDWallet()
         self.base_puzzle_program = None
         self.base_inner_puzzle_hash = None
         self.standard_wallet = wallet
         self.log = logging.getLogger(name if name else __name__)
+        self.log.info("Creating DID wallet from recovery file ...")
         self.wallet_state_manager = wallet_state_manager
-        self.did_info = DIDInfo(None, [], uint64(0), [], None, None, None, None, False)
+        self.did_info = DIDInfo(None, [], uint64(0), [], None, None, None, None, False, "")
         info_as_string = json.dumps(self.did_info.to_json_dict())
         self.wallet_info = await wallet_state_manager.user_store.create_wallet(
-            "DID Wallet", WalletType.DISTRIBUTED_ID.value, info_as_string
+            name, WalletType.DISTRIBUTED_ID.value, info_as_string, in_transaction=True
```

Comment:
Is the `in_transaction=True` correct here? Where is the DB transaction being created? From what I can tell, `create_new_did_wallet_from_recovery` is only called by `wallet_rpc_api.py` which doesn't seem to create a DB transaction prior to calling this function. I see the wallet_state_manager's lock being acquired, but that's not the same lock as used by the wallet_user_store's DBWrapper.